src/Utils/StrictPlus.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\Callback;
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class for better experience.
+ * - 'did you mean' hints
+ * - access to undeclared members throws exceptions
+ * - support for @property annotations
+ * - support for calling event handlers stored in $onEvent via onEvent()
+ * - compatible with Nette\Object
+ */
+trait StrictPlus
```

Comment:
Can anyone think of better name for this? The _plus_part implies even more strict behavior, but it's not the case, instead it adds totally non-strict syntactic sugar...